sugar-cat7/vspo-portal
# Reduce Next.js warnings

Closes #240.

**What this PR solves / how to test:**
This PR eliminates the Next.js warnings noted in #240.
1. Removed usage of `Image.layout` and used the `Image.style` prop instead to control the image size.
2. Replaced `images.domains` in the Next.js config with `images.remotePatterns`.
3. Added the `sharp` package for image optimization.

We still have a lot of warnings regarding the use of `Image.fill` (screenshot below), so I also tried to reduce the number of these warnings by removing the Image component from avatars (https://github.com/sugar-cat7/vspo-portal/commit/b466cd0b64babe4c063cf6d5d49f03b6d1cae724).
I can revert this if you suspect this hurts performance.

<img src="https://github.com/sugar-cat7/vspo-portal/assets/155891765/ea2acc71-44f2-452b-a80f-15e4ddcdfaee" width="400px">

To test:
- On `dev`, opening `/about` (or any page) should not log any warnings to the web console regarding the use of the legacy `Image.layout` prop.
- When `start`ing the app, there should be no warning in the command line regarding the use of the deprecated config property `images.domains`.
- When `start`ing the app, there should be no warning in the command line prompting the user to install `sharp`.
- All images in the app should appear as they did before.
  - Images on `/about` should be displayed with the same dimensions as before.
  - No image with a valid URL should be blocked ([throwing a 400 error](https://nextjs.org/docs/pages/api-reference/components/image#remotepatterns) in the web console) from being shown.
